src/abstractions/Microsoft.Kiota.Abstractions.csproj

```diff
@@ -19,6 +19,6 @@
 
   <!-- NET 5 target to be removed on next major version-->
   <ItemGroup Condition="'$(TargetFramework)' == 'net5.0' or '$(TargetFramework)'== 'netStandard2.0' or '$(TargetFramework)' == 'netStandard2.1'">
-      <PackageReference Include="System.Diagnostics.DiagnosticSource" Version="[6.0,9.0)" />
+      <PackageReference Include="System.Diagnostics.DiagnosticSource" Version="[6.0,10.0)" />
```

Comment:
```suggestion
      <PackageReference Include="System.Diagnostics.DiagnosticSource" Version="[6.0,)" />
```

Reply:
unbounded is risky in case they ever ship a breaking change impacting the APIs we use. Keeping it bound means we need to bump the boundary every net release, but at least it gives us a change to test/fix things.

Reply:
Won't setting an upper bound introduce compatibility errors with dependent projects

Reply:
What @baywet mentions is the reason behind the upper bound. In the event an new version is shipped that is breaking we get the opportunity to deal with it before the releasing of the package as opposed to getting reports of breakages from the customers.

Reply:
> Won't setting an upper bound introduce compatibility errors with dependent projects

The range covers all available versions available until the next breaking change of the dependency. Yes?

Reply:
Just curious, do breaking changes happen that often with Microsoft libraries between .NET versions? If not, I'd say the risk isn't high

Reply:
yes they do, We're simply lucky that none of those have affected us to date.
https://learn.microsoft.com/en-us/dotnet/core/compatibility/9.0